### PR TITLE
pass bundle offsets in search

### PR DESCRIPTION
**Pull request by PSeitz**
### Context / purpose
Describe the context and purpose of this PR. Link related issue(s) if any.

### Description
Describe the proposed changes made in this PR.

### How was this PR tested?
Describe how you tested this PR.

### Checklist
*Remove or complete this list if required.*
- [ ] tested locally
- [ ] added unit tests
- [ ] included documentation

### CODE REVIEW

File: quickwit-proto/proto/search_api.proto [hunk 1]
```diff
@@ -109,7 +109,18 @@ message LeafSearchRequest {
...
+
+message LeafSearchRequestMetadata {
+  // The offset of the start of footer in the split bundle. The footer contains the file bundle metadata and the hotcache.
+  uint64 split_footer_start = 1;
+  // The offset of the end of the footer in split bundle. The footer contains the file bundle metada and the hotcache.
+  uint64 split_footer_end = 2;
+
+  // Index split id to apply the query on.
+  // This id is resolved from the index_uri defined in the search_request.
+  string split_id = 3;
```
Comment (fulmicoton) [inline quickwit-proto/proto/search_api.proto:123]: nitpick. can we split_id first and reassign the 1,2,3 codes.

File: quickwit-search/src/root.rs [hunk 2]
```diff
@@ -709,7 +718,12 @@ mod tests {
             .expect_leaf_search()
             .times(2)
             .returning(|leaf_search_req: quickwit_proto::LeafSearchRequest| {
-                if leaf_search_req.split_ids == vec!["split1".to_string()] {
+                let split_ids = leaf_search_req
+                    .split_metadata
+                    .iter()
+                    .map(|metadata| metadata.split_id.to_string())
+                    .collect_vec();
+                if split_ids == vec!["split1".to_string()] {
```
Comment (fulmicoton) [inline quickwit-search/src/root.rs:726]: Vec on the left hand is useless here.

```&split_ids == &["split1".to_string()]```

If we keep doing it in test, we'll end up doing it in code.

File: quickwit-search/src/root.rs [hunk 3]
```diff
@@ -719,7 +733,7 @@ mod tests {
                         failed_splits: Vec::new(),
                         num_attempted_splits: 1,
                     })
-                } else if leaf_search_req.split_ids == vec!["split2".to_string()] {
+                } else if split_ids == vec!["split2".to_string()] {
```
Comment (fulmicoton) [inline quickwit-search/src/root.rs:736]: Same here

**Review by fulmicoton — APPROVED** [review]
LGTM have a look at the comments though

**Comment by PSeitz** [comment]
I'll address the comments in the next PR

**Pull request merged**